src/components/events/EventsSection.jsx

```diff
@@ -48,18 +48,26 @@ class EventsSection extends React.Component {
   render() {
     const { events } = this.state;
     const headings = ['Event Title', 'Event Date', 'No. Dancers', 'No. Performances', 'No. Judges'];
+    const showEvents = Array.isArray(events) && events.length > 0;
     return (
       <React.Fragment>
         <SectionHeader title="event" />
         <Table>
           <TableHeader headings={headings} />
           <TableBody>
-            {(Array.isArray(events) && events.length)
-              && events.map(event => (<EventTableRow key={event.id} {...event} />))
-            }
+            {showEvents && events.map((event) => {
+              const keys = ['eventTitle', 'eventDate', 'numJudges'];
+              const currentValues = pick(event, keys);
+              return (
+                <EventTableRow
+                  currentValues={currentValues}
+                  id={event.id}
+                  key={event.id} />);
+            })
+          }
           </TableBody>
         </Table>
-        {!events && (
+        {!showEvents && (
```

Comment:
Shouldn't it be `!showEvents || events.length == 0`? Otherwise, we just get an empty page :(